elastic/kibana
# [Discover] Fix saved search hidden chart can't be opened when returning to Discover

## Summary

Fixes #122555

This PR changes how `hideChart` option updates in state.

### Test notes

- Open discover, click "Chart options", and click "Hide chart".
- Click save button and follow save dialog to create saved search.
- Click "Chart options" and click "Open chart"
- Use left menu to open another application, like Dashboard
- Use left menu to open Discover, chart should be opened
- Click "Chart options" and click "Hide chart". Show/hide button should works after navigating back to discover.


### Checklist

Delete any items that are not applicable to this PR.
